### PR TITLE
[12.0][MIG] Migrated product_code_mandatory to v12

### PR DESCRIPTION
**Pull request by sudhir-erpharbor**
Please review and give feedback.

### CODE REVIEW

**Review by alan196 — APPROVED** [review]
Code review 👍

File: product_code_mandatory/README.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
+.. image:: https://img.shields.io/badge/license-AGPL--3-blue.png
```
Comment (tbaden) [inline product_code_mandatory/README.rst:1]: please use the new split readme [maintainer-tools](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0) and generate readme from it.

File: product_code_mandatory/views/product_view.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        <field name="inherit_id" ref="product.product_template_only_form_view"/>
+        <field name="arch" type="xml">
+            <field name="default_code" position="attributes">
+                <attribute name="attrs">{'invisible': [('product_variant_count', '&gt;', 1)],
+                    'required': [('product_variant_count', '=', 1)]}</attribute>
+            </field>
+        </field>
+    </record>
+
+    <record id="product_normal_form_code_mandatory_view" model="ir.ui.view">
```
Comment (tbaden) [inline product_code_mandatory/views/product_view.xml:16]: aesthetic change (optional): use original_id

>  In the case where an inherited view has a name which does not follow the guidelines set above, prefer naming the inherited view after the original over using a name which follows the guidelines.

File: product_code_mandatory/views/product_view.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+
+    <record id="product_template_only_form_code_mandatory_view" model="ir.ui.view">
```
Comment (tbaden) [inline product_code_mandatory/views/product_view.xml:4]: aesthetic change (optional): use original_id

>  In the case where an inherited view has a name which does not follow the guidelines set above, prefer naming the inherited view after the original over using a name which follows the guidelines.

**Review by tbaden — CHANGES_REQUESTED** [review]

File: product_code_mandatory/README.rst (hunk 1 above)
Comment (tbaden) [reply product_code_mandatory/README.rst:1]: fragmented readme, bot will generate the readme file

**Review by tbaden — APPROVED** [review]

**Comment by tbaden** [comment]
@sudhir-erpharbor please squash your commits into one and it's good to go

**Review by andreampiovesana — APPROVED** [review]
merge ok!

**Pull request merged**